www/public/repl/repl.js

```diff
@@ -165,11 +165,48 @@ async function processInputQueue() {
 // Callbacks to JS from Rust
 // ----------------------------------------------------------------------------
 
+var ROC_PANIC_INFO = null;
+
+function send_panic_msg_to_js(rocstr_ptr, panic_tag) {
+    const { memory } = repl.app.exports;
+
+    const rocStrBytes = new Int8Array(memory.buffer, rocstr_ptr, 12);
+    const finalByte = rocStrBytes[11]
+
+    let stringBytes = "";
+    if (finalByte < 0) {
+        // small string
+        const length = finalByte ^ 0b1000_0000;
```

Comment:
There's a bug here!

I did the following test in my browser console.
Let's simulate a Roc string of length 5 so the last byte is 0x85. Let's put that number into an Int8Array and take it out again
```js
a = new Int8Array([0x85])
finalByte = a[0] // -123
length = finalByte ^ 0b1000_0000 // -251
correctLength = finalByte + 128 // 5
```
Once you take it out of the Int8Array as a single `number` it is no longer 8 bits. It's converted to a JavaScript `number`. Bitwise ops in JS are effectively signed 32 bit. Bitwise ops on negative JS numbers are annoying, but addition just does what you'd expect. So `finalByte + 128` will work correctly.

Reply:
love JS numbers... fixed now